src/data/lex.rs

```diff
@@ -231,8 +231,9 @@ pub enum Token {
 
     // Misc
     Ellipsis,
-    StructDeref, // ->
-    Hash,        // #, used for preprocessing
+    StructDeref,    // ->
+    Hash,           // #, used for preprocessing
+    HashHash(bool), // ##, used for preprocessing (the bool is true unless it is created by `# ## #`)
```

Comment:
It's not clear what the bool is where you use it in the code, might be better to say explicitly (the comment is fine though).
```suggestion
    HashHash { operator: bool }, // ##, used for preprocessing (the bool is true unless it is created by `# ## #`)
```